devnet-sdk/testing/systest/multi_client.go

```diff
@@ -186,15 +186,66 @@ func (mc *MultiClient) fetchWithConsistencyCheck(
 		return nil, err
 	}
 
+	// Get the state root for the primary
+	primaryHeader, ok := primaryItem.(*types.Header)
+	if !ok {
+		// If we're fetching a block instead of a header
+		if block, ok := primaryItem.(*types.Block); ok {
+			primaryHeader = block.Header()
+		} else {
+			return nil, fmt.Errorf("expected header or block, got %T", primaryItem)
```

Comment:
Outside of the scope of this PR, but because we are using an `interface{}` type we end up with pretty painful code like this trying to figure out which concrete type we ended up with. Using `interface{}` is mostly an antipattern and seems unnecessary here since we know we will get something which is a block header or trivially convertible to a block header. 

This could definitely be improved in follow up work cc @teddyknox.

Reply:
Agreed. It feels like a refactor could simplify this code a lot.

Reply:
Fixed in #16069, which aims to merge into this PR branch.